assertion/affiliation/affiliation.go

```diff
@@ -178,48 +178,51 @@ func (a *Affiliation) computeTriggersForCastingSites(pass *analysis.Pass, upstre
 					}
 
 				case *ast.CompositeLit:
-					nodeType := util.TypeOf(pass, node.Type)
-
-					// If the composite is initializing a map, then check for possible pseudo-assignments
-					// for keys and values. For instance, if the type of value of the map is an interface
-					// but a struct is added to the map.
-					if mpType, ok := nodeType.(*types.Map); ok {
-						elemType := mpType.Elem()
-						keyType := mpType.Key()
-						// Iterate through each element in the composite
+					switch nodeType := node.Type.(type) {
+					case *ast.ArrayType:
+						// A slice (or array) declared of type interface, and initialized with a struct
+						// e.g., _ = []I{&S{}}
```

Comment:
Can we add some comments stating the current limitations (i.e., https://github.com/uber-go/nilaway/issues/46)? :) I had a feeling that we may need help from the type-checker in the future since Go allows you omit some type parts on syntactical level....

Reply:
Good point. Added a comment linking the issue.